applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py

```diff
@@ -38,19 +38,13 @@ def __init__(self, optimization_settings, optimization_mdpa):
             from mesh_controller_basic_updating import MeshControllerBasicUpdating
             self.mesh_controller = MeshControllerBasicUpdating(optimization_mdpa)
 
-    # --------------------------------------------------------------------------
-    def IsOptimizationModelPartAlreadyImported(self):
-        if self.optimization_mdpa.NumberOfNodes()>0:
-            self.__CheckIfDomainSizeIsSet()
-            return True
-        else:
-            return False
-
     # --------------------------------------------------------------------------
     def ImportOptimizationModelPart(self):
-        model_part_io = ModelPartIO(self.optimization_settings["design_variables"]["optimization_model_part_name"].GetString())
-        model_part_io.ReadModelPart(self.optimization_mdpa)
-        self.__CheckIfDomainSizeIsSet()
+        if self.__IsOptimizationModelPartAlreadyImported():
+            print("> Skipping import of optimization model part as already done by another application. ")
```

Comment:
I recommend to use the logger

Reply:
I would want to introduce the logger separately to not blow up this PR any further.